### PR TITLE
[Gluten-129] Adding github actions framework

### PR DESCRIPTION
**Pull request by zhouyuan**
This patch adds GHA for below tests:
* each PR should be prefixed with Gluten-: e.g Gluten-12
* format check on C++ code
* trigger unit tests for each PR

Signed-off-by: Yuan Zhou <yuan.zhou@intel.com>

### CODE REVIEW

File: .github/workflows/dev_cron/title_check.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+-->
+
+Thanks for opening a pull request!
+
+Could you open an issue for this pull request on Github Issues?
+
+https://github.com/oap-project/native-sql-engine/issues
```
Comment (zhouyuan) [inline .github/workflows/dev_cron/title_check.md:24]: fix url

File: .github/workflows/dev_cron/title_check.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+Then could you also rename ***commit message*** and ***pull request title*** in the following format?
+
+    [GJ-${ISSUES_ID}] ${detailed message}
+
+See also:
+
+  * [Other pull requests](https://github.com/oap-project/native-sql-engine/pulls/)
```
Comment (zhouyuan) [inline .github/workflows/dev_cron/title_check.md:32]: ditto

File: .github/workflows/unittests.yml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  formatting-check:
+    name: Formatting Check
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Run clang-format style check for C/C++ programs.
+      uses: jidicula/clang-format-action@v3.5.1
+      with:
+        clang-format-version: '10'
+        check-path: 'native-sql-engine/cpp/src'
```
Comment (zhouyuan) [inline .github/workflows/unittests.yml:71]: fix layout

**Comment by zhouyuan** [comment]
@rui-mo @weiting-chen

**Comment by zhouyuan** [comment]
The GHA will work after merge - here's the example 
image

**Comment by rui-mo** [comment]
Pushed one commit to the used Velox branch to make the compilation workable.

**Comment by weiting-chen** [comment]
It looks good to me.

**Pull request merged**